django_short_urls/tests/new_test.py

```diff
@@ -12,79 +15,116 @@
+            'long_url': self.long_url,
+            'short_path': 'inva/lid'
+        }
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_BAD_REQUEST)
 
-        new(self.factory.post('/new', self.data))
+        data['allow_slashes_in_prefix'] = True
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_BAD_REQUEST)
 
-        self.data['long_url'] = 'http://some.thi.ng/different'
+        # Test for slashes in the prefix
+        data = {
+            'long_url': self.long_url,
+            'short_path': 'valid',
+            'prefix': 'inva/lid'
+        }
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_BAD_REQUEST)
 
-        self.assertEqual(new(self.factory.post('/new', self.data)).status_code, HTTP_CONFLICT)
+        # Test for other invalid chars in the url ('%', '?', etc.)
+        data = {
+            'long_url': self.long_url,
+            'short_path': 'inva&lid',
+        }
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_BAD_REQUEST)
 
-        self.data['prefix'] = 'foobar'
+    def test_short_path_conflict(self):
+        # Create the base conflicting link
+        data = {
+            'long_url': self.long_url,
+            'short_path': 'conflict'
+        }
+        new(self._post('/new', data))
 
-        new(self.factory.post('/new', self.data))
+        # Conflict when trying to replace an existing short_path with a different long_url target
+        data['long_url'] = 'http://some.thi.ng/different'
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_CONFLICT)
 
-        self.data['long_url'] = 'http://yet.aga.in/something/else'
+        # Now check the same thing with urls that include a prefix - first create
+        data['prefix'] = 'foobar'
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_OK)
 
-        self.assertEqual(new(self.factory.post('/new', self.data)).status_code, HTTP_CONFLICT)
+        # Now trigger the conflict
+        data['long_url'] = 'http://yet.aga.in/something/else'
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_CONFLICT)
 
     def test_forbidden_keyword(self):
-        self.data['long_url'] = 'http://work4.com'
-        self.data['short_path'] = 'jobs'
-
-        self.assertEqual(new(self.factory.post('/new', self.data)).status_code, HTTP_FORBIDDEN)
-
-    def test_new(self):
-        self.data['long_url'] = 'http://www.work4labs.com/'
-
-        self.assertEqual(new(self.factory.post('/new', self.data)).status_code, HTTP_OK)
-
-        self.data['short_path'] = 'bar'
-        self.data['prefix'] = 'foo'
+        data = {
+            'long_url': self.long_url,
+            'short_path': 'jobs'
+        }
+        self.assertEqual(new(self._post('/new', data)).status_code, HTTP_FORBIDDEN)
 
-        self.assertEqual(new(self.factory.post('/new', self.data)).status_code, HTTP_OK)
+    @patch('django_short_urls.models.statsd')
```

Comment:
What about testing if statsd was actually called in those cases? Are we assuming it's already tested in `models_link_test.py`?